util/named_requirement_verification/legacy_forward_iterator.h

```diff
@@ -0,0 +1,168 @@
+   * verification
+   *
+   * @tparam It Type of iterator for verification
+   * @return std::pair<bool,array<string_view>> First represents
+   * satisfaction of the requirement. Second contains error messages
+   */
+  template <typename It>
+  auto is_satisfied_for(It valid_iterator, const size_t container_size) {
+    auto legacy_input_iterator_res =
+        legacy_input_iterator_requirement{}.is_satisfied_for<It>();
+    if (legacy_input_iterator_res.first == false) {
+      m_errors.add_errors(legacy_input_iterator_res.second);
+    }
+
+    constexpr bool is_dereferenceable = is_dereferenceable_v<It>;
+    constexpr bool can_pre_increment =
+        type_traits::has_arithmetic::pre_increment_v<It>;
+    constexpr bool can_post_increment =
+        type_traits::has_arithmetic::post_increment_v<It>;
+    constexpr bool has_reference_member =
+        type_traits::has_field::reference_v<It>;
+    constexpr bool has_value_type_member =
+        type_traits::has_field::value_type_v<It>;
+    constexpr bool has_equal_operator =
+        type_traits::has_comparison::is_equal_v<It>;
+    constexpr bool is_def_constructable = std::is_default_constructible_v<It>;
+
+    if (!is_def_constructable) {
+      m_errors.add_error("Iterator have to be default constructible");
+    }
+
+    using it_traits = std::iterator_traits<It>;
+
+    // Allows us to use reference_t from iterator_traits
+    if constexpr (has_reference_member) {
+      const bool is_output_iterator_req_satisfied =
+          legacy_output_iterator_requirement{}.is_satisfied_for<It>().first;
+
+      if (is_output_iterator_req_satisfied) {
+        if (std::is_const_v<typename it_traits::reference> == true) {
+          m_errors.add_error(
+              "Provided iterator satisfy to LegacyOutputIterator "
+              "requirement. iterator_traits::reference have to be non const");
+        }
+      } else {
+        if (std::is_const_v<typename it_traits::reference> == false) {
+          m_errors.add_error(
+              "Provided iterator not satisfy to LegacyOutputIterator "
+              "requirement. iterator_traits::reference have to be const");
+        }
+      }
+    }
+
+    if constexpr (can_post_increment) {
+      if (std::is_same_v<decltype(std::declval<It&>()++), It> == false) {
+        m_errors.add_error("operator++(int) have to return It");
+      }
+    }
+
+    if constexpr (can_post_increment && is_dereferenceable &&
+                  has_reference_member) {
+      if (std::is_convertible_v<decltype(*(std::declval<It&>()++)),
+                                typename it_traits::reference> == false) {
+        m_errors.add_error(
+            "Expression *i++ have to be convertible to "
+            "iterator_traits::reference");
+      }
+    }
+
+    if (container_size == 0) {
+      m_errors.add_error(
+          "Some of the test requires container size more than 0. These tests "
+          "have been skipped");

```

Comment:
I'm not sure if using `m_errors` is the right approach here.
`m_errors` is designed to return an implementation errors, where this case looks like a SYCL-CTS bug.

Reply:
My concern is not the variable name, but rather the code itself. `container_size` is set by the test and not by implementation, so I don't see any point in this check. Why do you need it here? Why can't we set `container_size` to be always != 0?

Reply:
> Why can't we set `container_size` to be always != 0?

We can do this, I've done this [here](https://github.com/KhronosGroup/SYCL-CTS/pull/354/commits/7f1f1f7f6a2153f59266ed9fe8057bd807d69c91#diff-e651b129eae9f732f1c5dec2b81da878eb24aee84c8c9da0b054707fd1867285L73-R70)

Reply:
The link points to the wrong commit and I still see the check for `container_size != 0`  here: https://github.com/KhronosGroup/SYCL-CTS/pull/354/files#diff-98e2bb40cc64308f2138f8199c7e0eaa1fc5eb582f99d3192586afe30c8fd2d0R84-R88.
This branch reports **an error** that test is **skipped**. This sounds wrong to me. Please, take a look how other tests communicate information about skipping test cases (e.g. using WARN macro).

Reply:
`WARN` macro has been added [here](https://github.com/KhronosGroup/SYCL-CTS/pull/354/commits/5892d5383288816890fa85b5405964543449d5f6)